### PR TITLE
[chore] [pdata] Add test coverage for slices Sort method

### PR DESCRIPTION
**Pull request by dmitryax**

### CODE REVIEW

File: pdata/internal/cmd/pdatagen/internal/base_slices.go [hunk 1]
```diff
@@ -240,6 +240,22 @@ func (es ${structName}) Sort(less func(a, b ${elementName}) bool) {
 	sort.SliceStable(*es.orig, func(i, j int) bool { return less(es.At(i), es.At(j)) })
 }`
 
+const slicePtrTestTemplate = `func Test${structName}_Sort(t *testing.T) {
+	es := generateTest${structName}()
+	es.Sort(func(a, b ${elementName}) bool {
+		return uintptr(unsafe.Pointer(a.orig)) < uintptr(unsafe.Pointer(b.orig))
+	})
+	for i := 1; i < es.Len(); i++ {
+		assert.True(t, uintptr(unsafe.Pointer(es.At(i-1).orig)) < uintptr(unsafe.Pointer(es.At(i).orig)))
```
Comment (bogdandrutu) [inline pdata/internal/cmd/pdatagen/internal/base_slices.go:249]: assert.Less?

File: pdata/internal/cmd/pdatagen/internal/base_slices.go [hunk 2]
```diff
@@ -240,6 +240,22 @@ func (es ${structName}) Sort(less func(a, b ${elementName}) bool) {
...
+	}
+	es.Sort(func(a, b ${elementName}) bool {
+		return uintptr(unsafe.Pointer(a.orig)) > uintptr(unsafe.Pointer(b.orig))
+	})
+	for i := 1; i < es.Len(); i++ {
+		assert.True(t, uintptr(unsafe.Pointer(es.At(i-1).orig)) > uintptr(unsafe.Pointer(es.At(i).orig)))
```
Comment (bogdandrutu) [inline pdata/internal/cmd/pdatagen/internal/base_slices.go:255]: same.

File: pdata/internal/cmd/pdatagen/internal/base_slices.go (hunk 1 above)
Comment (dmitryax) [reply pdata/internal/cmd/pdatagen/internal/base_slices.go:249]: Surprisingly assert.Less doesn't support `uintptr`

Comment (bogdandrutu) [reply pdata/internal/cmd/pdatagen/internal/base_slices.go:249]: Trying to fix that :) https://github.com/stretchr/testify/pull/1339

Comment (bogdandrutu) [reply pdata/internal/cmd/pdatagen/internal/base_slices.go:249]: Maybe add a TODO to change this if the PR gets merged.

**Review by bogdandrutu — APPROVED** [review]

File: pdata/internal/cmd/pdatagen/internal/base_slices.go (hunk 1 above)
Comment (dmitryax) [reply pdata/internal/cmd/pdatagen/internal/base_slices.go:250]: Done

**Pull request merged**